plugins/bigcommerce/src/data-plugin.ts

```diff
@@ -0,0 +1,160 @@
+    let path = resource;
+    if (path === "product") {
+        path = "products";
+    } else if (path === "category") {
+        path = "categories";
+    }
+    if (resourceId) {
+
+        path += `/${resourceId}`;
+    }
+
+    // Initialize URLSearchParams object to construct query parameters
+    const params = new URLSearchParams();
+
+    // Set 'limit' parameter
+    if (limit) {
+        params.set('limit', limit.toString());
+    }
+
+    // Interpret and set 'query' parameter
+    if (query) {
+        params.set('keyword', query);
+    }
+
+    // Construct the full URL
+    return `${base}/${path}?${params.toString()}`;
+}
+
+
+// const buildPath = ({
+//     resource,
+//     resourceId,
+// }: {
+//     resource: BigCommerceResourceType;
+//     resourceId?: string;
+// }) => {
+//     // BigCommerce API paths for products and categories
+//     switch (resource) {
+//         case 'product':
+//             return resourceId ? `products/${resourceId}` : 'products';
+//         case 'category':
+//             return resourceId ? `categories/${resourceId}` : 'categories';
+//         default:
+//             throw new Error('Unsupported resource type');
+//     }
+// };
+
+
+const RESOURCE_TYPES: {
+    name: string;
+    id: BigCommerceResourceType;
+    description: string;
+}[] = [
+        {
+            name: 'Product',
+            id: 'product',
+            description: 'All of your BigCommerce products.',
+        },
+        {
+            name: 'Category',
+            id: 'category',
+            description: 'All of your BigCommerce categories.',
+        },
+    ];
+
+interface DataPluginConfig extends APIOperations {
+    name: string;
+    icon: string;
+}
+
+export const getDataConfig = (service: CommerceAPIOperations): DataPluginConfig => {
+    return {
+        name: 'BigCommerce',
+        icon: 'https://iili.io/JnTyc4s.png',
+        getResourceTypes: async () =>
+            RESOURCE_TYPES.map(
+                (model): ResourceType => ({
+                    ...model,
+                    inputs: () => [
+                        {
+                            friendlyName: 'limit',
+                            name: 'limit',
+                            type: 'number',
+                            defaultValue: 10,
+                            max: 250, // Reflecting BigCommerce's API limit
+                            min: 1,
+                        },
+                        {
+                            friendlyName: 'Search',
+                            name: 'query',
+                            type: 'string',
+                        },
+                    ],
+
+                    toUrl: ({ entry, query, limit }) =>
+                        //this request fails due to lack of auth headers but doesn't seem to affect the functionality
+                        buildBigCommerceUrl({
```

Comment:
The url has to work for the data to be fetched and included in content, you can append headers to proxied calls same as this function https://github.com/BuilderIO/builder/blob/539f435a3a10f8b44e517e6f794dfc2c6261784d/plugins/contentstack/src/plugin.ts#L218